setup.sh

```diff
@@ -17,8 +31,10 @@ check_install_python3(){
 
 check_install_python3
 
-# On debian based systems, --user flag is used by default when not running in virtualenv or as root
-pip install -r requirements.txt
+check_install_module "pip" "python3-pip" 
+
+#On debian based systems, --user flag is used by default when not running in virtualenv or as root
+python3 -m pip install -r requirements.txt
 
```

Comment:
Could you check if sphinx is callable here? If it is installed into `~/.local/bin` and that is not in `PATH`, then the Makefile won't work... So, it would be good to notify the user to add it to the PATH (e.g. in .profile / .bashrc / or some similar place) and exit if sphinx is not available here.